website/versioned_docs/version-experimental/RelayHooks-StepByStep.md

```diff
@@ -0,0 +1,300 @@
+      console.error(error);
+    });
+
+    return () => {
+      isMounted = false;
+    };
+  }, [fetchGraphQL]);
+
+  // Render "Loading" until the query completes
+  return (
+    <div className="App">
+      <header className="App-header">
+        <p>
+          {name != null ? `Repository: ${name}` : "Loading"}
+        </p>
+      </header>
+    </div>
+  );
+}
+
+export default App;
+```
+
+## Step 3: When To Use Relay
+
+At this point we can fetch data with GraphQL and render it with React. This is a reasonable solution that can be sufficient for many apps. However, this approach doesn't necessarily scale. As our app grows in size and complexity, or the number of people working on the app grows, a simple approach like this can become limiting. Relay provides a number of features designed to help keep applications fast and reliable even as they grow in size and complexity: colocating data dependencies in components with GraphQL fragments, data consistency, mutations, etc. Check out [Thinking in GraphQL](/docs/en/thinking-in-graphql.html) and [Thinking in Relay](/docs/en/thinking-in-relay.html) for an overview of how Relay makes it easier to work with GraphQL.
+
+
+## Step 4: Adding Relay To Our Project
+
+In this guide we'll demonstrate installing the *experimental* release of Relay Hooks, a new, hooks-based Relay API that supports React Concurrent Mode and Suspense. 
+
+First we'll add the necessary packages. Note that Relay is comprised of three key pieces: a compiler (which is used at build time), a core runtime (that is React agnostic), and a React integration layer.
+
+```bash
+# NPM Users
+npm install --save relay-runtime react-relay@experimental
+npm install --save-dev relay-compiler graphql
+
+# Yarn Users
+yarn add relay-runtime react-relay@experimental
+yarn add --dev relay-compiler graphql babel-plugin-relay
+```
+
+*Note*: You may get a notice asking you to choose which version of `relay-runtime` to use - if so, specify version 7.0.x (e.g. `7.0.0`).
+
+### 4.1 Configure Relay Compiler
+
+Next let's configure Relay compiler. We'll need a copy of the schema as a `.graphql` file. If you're using the GitHub GraphQL API, you can download a copy directly from the Relay example app:
+
+```bash
+cd your-app-name
+curl https://raw.githubusercontent.com/relayjs/relay-examples/master/issue-tracker/schema/schema.graphql > schema.graphql
+```
+
+If you're using your own API we suggest using the [`get-graphql-schema`](https://www.npmjs.com/package/get-graphql-schema) utility to download your schema into a `.graphql` file.
+
+Now that we have a schema we can modify `package.json` to run the compiler first whenever we build or start our app:
+
+```json
+// your-app-name/package.json
+{
+  ...
+  "scripts": {
+    ...
+    "start": "yarn run relay && react-scripts start",
+    "build": "yarn run relay && react-scripts build",
+    "relay": "yarn run relay-compiler --schema schema.graphql --src ./src/ --watchman false $@"
```

Comment:
so we're explicitly not running watch mode here, right?

Reply:
yes bc people may not have watchman installed,  i  comment about it  below